Mono.Debugger.Soft/Mono.Debugger.Soft/ObjectMirror.cs

```diff
@@ -142,37 +141,43 @@ public class ObjectMirror : Value {
 			}
 		}
 
-		public Value InvokeMethod (ThreadMirror thread, MethodMirror method, IList<Value> arguments) {
-			return InvokeMethod (vm, thread, method, this, arguments, InvokeOptions.None);
+		Value IInvocableMethodOwnerMirror.GetThisObject () {
+			return this;
 		}
 
-		public Value InvokeMethod (ThreadMirror thread, MethodMirror method, IList<Value> arguments, InvokeOptions options) {
-			return InvokeMethod (vm, thread, method, this, arguments, options);
+		void IInvocableMethodOwnerMirror.ProcessResult (InvokeResult result)
+		{
+		}
+
+		// Public invocation API
+
+		public static Value InvokeMethod (IInvocableMethodOwnerMirror mirror, ThreadMirror thread, MethodMirror method, IList<Value> arguments, InvokeOptions options = InvokeOptions.None) {
+			return InvokeMethod (mirror, mirror.VirtualMachine, thread, method, mirror.GetThisObject (), arguments, options);
 		}
 
 		[Obsolete ("Use the overload without the 'vm' argument")]
-		public IAsyncResult BeginInvokeMethod (VirtualMachine vm, ThreadMirror thread, MethodMirror method, IList<Value> arguments, InvokeOptions options, AsyncCallback callback, object state) {
-			return BeginInvokeMethod (vm, thread, method, this, arguments, options, callback, state);
+		public static IAsyncResult BeginInvokeMethod (IInvocableMethodOwnerMirror mirror, VirtualMachine vm, ThreadMirror thread, MethodMirror method, IList<Value> arguments, InvokeOptions options, AsyncCallback callback, object state) {
```

Comment:
It's a reason. But do you support backward ABI compat between major releases of XS?
I can keep this ~10 methods as instance members but I guess it's really bad design, because if you need to implement invocations for one another type, you have to copy paste these 10 methods.